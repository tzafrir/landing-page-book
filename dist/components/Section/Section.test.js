'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

var _enzymeAdapterReact = require('enzyme-adapter-react-16');

var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);

var _Section = require('./Section');

var _Section2 = _interopRequireDefault(_Section);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

(0, _enzyme.configure)({ adapter: new _enzymeAdapterReact2.default() });

describe('Section', function () {
  var props = void 0;
  var mounted = void 0;
  var section = function section() {
    if (!mounted) {
      mounted = (0, _enzyme.shallow)(_react2.default.createElement(_Section2.default, props));
    }
    return mounted;
  };
  describe('Section', function () {
    beforeEach(function () {
      props = {
        gray: false,
        title: 'Front10',
        subTitle: 'Awesome team',
        className: '',
        contentClassName: '',
        children: null
      };
      mounted = undefined;
    });

    it('should render', function () {
      expect((0, _enzyme.shallow)(_react2.default.createElement(_Section2.default, props))).toMatchSnapshot();
    });

    it('always renders a div', function () {
      var divs = section().find('div');

      expect(divs.length).toBeGreaterThan(0);
    });
  });
});