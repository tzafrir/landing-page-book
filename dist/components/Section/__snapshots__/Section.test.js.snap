// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Section should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Section
    alignHeader="center"
    className=""
    contentClassName=""
    gray={false}
    subTitle="Awesome team"
    title="Front10"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="p-5 "
      >
        <div
          className="mb-5 text-center"
        >
          <Header
            borderBottom={false}
            borderTop={false}
            className="Section__Title mb-2"
            color=""
            icon=""
            textAlign="center"
            type="h2"
          >
            Front10
          </Header>
          <Header
            borderBottom={false}
            borderTop={false}
            className="Section__Subtitle mb-2"
            color=""
            icon=""
            textAlign="center"
            type="h3"
          >
            Awesome team
          </Header>
        </div>
      </div>,
      "className": "Section  ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="mb-5 text-center"
          >
            <Header
              borderBottom={false}
              borderTop={false}
              className="Section__Title mb-2"
              color=""
              icon=""
              textAlign="center"
              type="h2"
            >
              Front10
            </Header>
            <Header
              borderBottom={false}
              borderTop={false}
              className="Section__Subtitle mb-2"
              color=""
              icon=""
              textAlign="center"
              type="h3"
            >
              Awesome team
            </Header>
          </div>,
          null,
        ],
        "className": "p-5 ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Header
                borderBottom={false}
                borderTop={false}
                className="Section__Title mb-2"
                color=""
                icon=""
                textAlign="center"
                type="h2"
              >
                Front10
              </Header>,
              <Header
                borderBottom={false}
                borderTop={false}
                className="Section__Subtitle mb-2"
                color=""
                icon=""
                textAlign="center"
                type="h3"
              >
                Awesome team
              </Header>,
            ],
            "className": "mb-5 text-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "borderBottom": false,
                "borderTop": false,
                "children": "Front10",
                "className": "Section__Title mb-2",
                "color": "",
                "icon": "",
                "textAlign": "center",
                "type": "h2",
              },
              "ref": null,
              "rendered": "Front10",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "borderBottom": false,
                "borderTop": false,
                "children": "Awesome team",
                "className": "Section__Subtitle mb-2",
                "color": "",
                "icon": "",
                "textAlign": "center",
                "type": "h3",
              },
              "ref": null,
              "rendered": "Awesome team",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="p-5 "
        >
          <div
            className="mb-5 text-center"
          >
            <Header
              borderBottom={false}
              borderTop={false}
              className="Section__Title mb-2"
              color=""
              icon=""
              textAlign="center"
              type="h2"
            >
              Front10
            </Header>
            <Header
              borderBottom={false}
              borderTop={false}
              className="Section__Subtitle mb-2"
              color=""
              icon=""
              textAlign="center"
              type="h3"
            >
              Awesome team
            </Header>
          </div>
        </div>,
        "className": "Section  ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="mb-5 text-center"
            >
              <Header
                borderBottom={false}
                borderTop={false}
                className="Section__Title mb-2"
                color=""
                icon=""
                textAlign="center"
                type="h2"
              >
                Front10
              </Header>
              <Header
                borderBottom={false}
                borderTop={false}
                className="Section__Subtitle mb-2"
                color=""
                icon=""
                textAlign="center"
                type="h3"
              >
                Awesome team
              </Header>
            </div>,
            null,
          ],
          "className": "p-5 ",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Header
                  borderBottom={false}
                  borderTop={false}
                  className="Section__Title mb-2"
                  color=""
                  icon=""
                  textAlign="center"
                  type="h2"
                >
                  Front10
                </Header>,
                <Header
                  borderBottom={false}
                  borderTop={false}
                  className="Section__Subtitle mb-2"
                  color=""
                  icon=""
                  textAlign="center"
                  type="h3"
                >
                  Awesome team
                </Header>,
              ],
              "className": "mb-5 text-center",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "borderBottom": false,
                  "borderTop": false,
                  "children": "Front10",
                  "className": "Section__Title mb-2",
                  "color": "",
                  "icon": "",
                  "textAlign": "center",
                  "type": "h2",
                },
                "ref": null,
                "rendered": "Front10",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "borderBottom": false,
                  "borderTop": false,
                  "children": "Awesome team",
                  "className": "Section__Subtitle mb-2",
                  "color": "",
                  "icon": "",
                  "textAlign": "center",
                  "type": "h3",
                },
                "ref": null,
                "rendered": "Awesome team",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          null,
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
